test/integration/hotrestart_test.sh

```diff
@@ -4,13 +4,13 @@ set -e
 
 [[ -z "${ENVOY_BIN}" ]] && ENVOY_BIN="${TEST_RUNDIR}"/source/exe/envoy-static
 
-# TODO(htuch): Clean this up when Bazelifying the hot restart test below. At the same time, restore
-# some test behavior lost in #650, when we switched to 0 port binding - the hot restart tests no
-# longer check socket passing. See #654.
+# TODO(htuch): Clean this up when Bazelifying the hot restart test below.
```

Comment:
Yeah. I think we could have a simple C++ binary that links against test_environment.cc that uses the substitution methods provided there, rather than the adhoc sed below. That way we don't need to worry about keeping this special case in sync with `TestEnvironment`.